pkg/kademlia/peer_discovery.go

```diff
@@ -65,6 +71,12 @@ func (lookup *peerDiscovery) Run(ctx context.Context) error {
 					}
 
 					next, _ = lookup.queue.Closest()
+
+					// Send node on channel if it matches target
+					if next.GetId() == lookup.target.String() {
+						lookup.foundOne <- next
```

Comment:
Is it able to connect to at this point? Or would it still need to wait for the other node to connect to this node.

Similarly this can deadlock the system when no-one is recv on this channel.